mqtt_as/mqtt_as.py

```diff
@@ -601,11 +541,9 @@ async def _keep_connected(self):
                 await asyncio.sleep(1)
                 gc.collect()
             else:
-                self._sta_if.disconnect()
+                await self._interface.disconnect()
                 await asyncio.sleep(1)
-                try:
-                    await self.wifi_connect()
-                except OSError:
+                if not await self._interface.connect():  # TODO: switch to reconnect once PR final.
```

Comment:
Please could you explain this comment.

Reply:
The code in this PR is not modified (yet), I just separated it from the original mqtt_as.
For the new interface to work better, the disconnect and connect here should be replaced as a reconnect, because that's what mqtt_as is requesting here. This way interface implementations can decide wether a reconnect is really needed or not. For your resilient wifi connection it might make sense to reconnect the wlan on ever mqtt connection problem but an ethernet or other implementation might be able to check the connection state reliably enough and decide not to reconnect the interface. This will also be relevant for projects that have multiple applications that rely on the wlan (e.g. additional webserver, http push/get). [Note: detecting a bad wifi link would then still be possible by checking the rate of reconnect requests and if those exceed e.g. 5 requests in 10 seconds, the wlan interface reconnects. But that's up to the user/dev to implement.]
In the default implementation of the WLAN interface, reconnect will default to the code seen here: disconnect, wait 1 sec, connect.